api/auth.mjs

```diff
@@ -0,0 +1,154 @@
+import { BAD_REQUEST, UNAUTHORIZED, TYPE_JSON, INTERNAL, rand } from './defs.mjs'
+import { GET, withUser } from './router.mjs'
+import * as db from './db.mjs'
+
+const gql = async (query) => {
+  const text = await query.text()
+  try {
+    const { data, errors } = JSON.parse(text)
+    return { data, error: errors && errors[0].message }
+  } catch {
+    return { error: text }
+  }
+}
+
+const GUILD = '821033851153285170'
+const DISCORD = 'https://discordapp.com/api'
+GET.auth.discord = async ({ url }) => {
```

Comment:
This is how we register a route, here this function will be called everytime a `GET` request is send on `/api/auth/discord`